core/drivers/pl022_spi.c

```diff
@@ -180,11 +188,17 @@ static void pl022_txrx8(struct spi_chip *chip, uint8_t *wdat, uint8_t *rdat,
 	}
 
 	if (rdat) {
-		while (j < num_pkts) {
+		while ((j < num_pkts) &&
+			(read8(pd->base + SSPSR) & SSPSR_BSY)) {
 			if (read8(pd->base + SSPSR) & SSPSR_RNE)
 				/* rx 1 packet */
 				rdat[j++] = read8(pd->base + SSPDR);
 		}
+
+		if (j < num_pkts) {
+			EMSG("# packets requested: %zu\n", num_pkts);
+			EMSG("# packets received: %zu\n", j);
```

Comment:
This looks more informative message than an error message since nothing fails (no error status propagated). If `pl022_txrx8()` can fail, shouldn't there rather be a return value for this? 
Same for `pl022_txrx16()` below.

minor: rather trace this over a single string line.